### DISCUSSION TITLE
Long press for dimming Hue High Bulb E27

### DISCUSSION THREAD

**Discussion by cwegh** [joleys/niko-home-control-II#192] [Q&A] [answered]
So I am using a virtual output in NHC linked to an action and a button to turn on and off a Hue Light Bulb E27 with a physical button of NHC (and returning also state to hue and vice versa). Works great. 

However I was wondering if it is also possible to pick up a long press of the physical NHC button to dim the Hue light bulb?

**Comment by tijsverkoyen** [comment, marked as answer]
Unfortunately this can not be achieved. The presses on buttons are not something that can be triggered on through the API.

**Comment by myllen** [comment]
I have managed to achieve this by putting your NHC button in pulse mode, enabling shift function on this button and creating a few automations in Home Assistant.

In NHC: Put your button in pulse mode. Create 2 virtual flags (let's call them VIRT1 & VIRT2).
Normal routine: push of the button,  start behaviour: VIRT1 to on, stop behaviour: none. (You want VIRT1 to latch on when pressed, so it stays on when button is released)
Shift routine: condition: if VIRT1 = on, start behaviour: VIRT2 to on, stop behaviour: VIRT2 to off. (You want VIRT2 to only stay on for the duration that the button is held/pushed)
If you have a feedback led on your button, set the behaviour to 'All devices have start behaviour value'.

In Home Assistant:

Automation 1:
    trigger: VIRT1 = on
    action: turn on lights
Automation 2:
    trigger: VIRT2 = on
    if VIRT2 is on for less than 1 second (simulating normal push), then turn off lights and set VIRT1 to off.
    else start dimming (set input boolean for dimming to on)
Automation 3:
    trigger: input boolean for dimming = on
    action: dimming logic
Automation 4:
    trigger: VIRT2 = off
    action: input boolean for dimming = off (ie stop dimming)
 

Technically, if you don't have a feedback led on your button, you could also do it with only 1 virtual flag (VIRT2 in this case) with a little more programming in HA. But I use VIRT1 to turn my feedback led to ON when the light is ON.
